### ISSUE TITLE
Rubocop gives false positive in a closure

### ISSUE THREAD

**Issue by dnagir** [bbatsov/rubocop#3896]
The code:
```
    class Stream
      def self.of(event_relation)
        Class.new(self) do
          def initialize(relation: event_relation, formatter: EventHubFormatter)
            super
          end
        end
      end
    end
```

is giving a warning:
`W: Unused method argument - event_relation. If it's necessary, use _ or _event_relation as an argument name to indicate that it won't be used. You can also write as of(*) if you want the method to accept any arguments but don't care about them.`

which is incorrect because the `event_relation` variable is used.

**Comment by dnagir** [comment]
Actually, Rubocop is correct here and this is an error in Ruby 🤕
